src/osp/Active/SysRender.cpp

```diff
@@ -138,24 +153,29 @@ RenderPipeline SysRender::create_forward_renderer()
             using Magnum::GL::Renderer;
             using namespace osp::active;
 
-            auto& reg = rScene.get_registry();
-
             Renderer::enable(Renderer::Feature::DepthTest);
             Renderer::enable(Renderer::Feature::FaceCulling);
             Renderer::enable(Renderer::Feature::Blending);
             Renderer::setBlendFunction(
-                Renderer::BlendFunction::SourceAlpha,
-                Renderer::BlendFunction::OneMinusSourceAlpha);
-
-            auto transparentView = reg.view<ACompTransparent, ACompVisible, ACompShader>();
-
-            // Draw backfaces
-            Renderer::setFaceCullingMode(Renderer::PolygonFacing::Front);
-            SysRender::draw_group(rScene, transparentView, camera);
-
-            // Draw frontfaces
-            Renderer::setFaceCullingMode(Renderer::PolygonFacing::Back);
-            SysRender::draw_group(rScene, transparentView, camera);
+                    Renderer::BlendFunction::SourceAlpha,
+                    Renderer::BlendFunction::OneMinusSourceAlpha);
+
+            // temporary: disabled depth writing makes the plumes look nice, but
+            //            can mess up other transparent objects once added
+            Renderer::setDepthMask(false);
```

Comment:
Not sure if I understand how this works without doing several passes?

Reply:
This makes the plumes just not write any new depth data, so they render on top of each other without care.

Reply:
Ah I see. It would be better to just sort the objects by depth and use the method I had above. The two-pass transparency is among the highest performance transparency rendering algorithms there is, the only issue is that it doesn't do per-pixel sorting for overlapping objects but that's an unsolved problem in computer graphics as far as I'm aware.